### PR TITLE
Add filter for cognito identity fields

### PR DESCRIPTION
**Pull request by dfangl**
This PR adds a hacky filter, to the cognito_identity field of the lambda context. This will only allow correct values to be passed into it.

Once a real fix is available (by rethinking and redoing the `localstack.services.apigateway.apigateway_listener.invoke_rest_api_integration_backend` and `localstack.services.apigateway.helpers.get_event_request_context` methods @whummer @calvernaz ) so that this part:
```python
        "identity": {
            "accountId": account_id,
            "sourceIp": source_ip,
            "userAgent": headers.get("User-Agent"),
        },
```
does not show up in the cognito_identity part of the lambda context, this PR should be reverted.

Until then, it will only allow `"cognitoIdentityId"` and `"cognitoIdentityPoolId"` as directory keys for cognito_identity.

This should help fixing #5783 .

Test for apigateway - rust lambda integration added to prevent further regressions.

### CODE REVIEW

File: localstack/services/awslambda/lambda_executors.py [hunk 1]
```diff
@@ -108,6 +108,9 @@
 # CWD folder of handler code in Lambda containers
 DOCKER_TASK_FOLDER = "/var/task"
 
+# TODO remove once clarification of apigateway contexts is complete. Really bad hack!!
```
Comment (dominikschubert) [inline localstack/services/awslambda/lambda_executors.py:111]: It's a good bad hack in the meantime though :smile:

**Review by dominikschubert — COMMENTED** [review]
Will leave the approval to the others, but from my side I  don't see an issue with having it in there as a temporary fix :+1:

**Review by calvernaz — APPROVED** [review]
I think it's an ok fix, the request context that is being to the event is being wrongly passed to the context of the lambda. I'll create an issue to complete the fix, by correctly passing the necessary information to build the content object.

File: localstack/services/awslambda/lambda_executors.py [hunk 2]
```diff
@@ -141,7 +144,11 @@ def __init__(
         self.invoked_function_arn = lambda_function.arn()
         if qualifier:
             self.invoked_function_arn += ":" + qualifier
-        self.cognito_identity = context.get("identity")
+        self.cognito_identity = context.get("identity") and {
```
Comment (whummer) [inline localstack/services/awslambda/lambda_executors.py:147]: nit: We could use `select_attributes(..)` from the `collections` utils here. 👍

**Review by whummer — APPROVED** [review]
Great catch and nice fix @dfangl ! 👍

File: localstack/services/awslambda/lambda_executors.py (hunk 2 above)
Comment (dfangl) [reply localstack/services/awslambda/lambda_executors.py]: done!

**Comment by dfangl** [comment]
Due to a lot of lambda integrations tested in there, I added test_apigateway.py to the tests executed for different lambda executors (new test is also in there). At some point we have to find a solution for this which scales better (pytest marks, decide on used fixtures etc).

**Review by thrau — APPROVED** [review]
LGTM :+1:

**Review by alexrashed — APPROVED** [review]
LGTM 🚀

**Pull request merged**